extensions/sql/asset/index/src/test/java/org/eclipse/dataspaceconnector/sql/asset/index/SqlAssetIndexTest.java

```diff
@@ -0,0 +1,260 @@
+import org.eclipse.dataspaceconnector.sql.pool.ConnectionPool;
+import org.eclipse.dataspaceconnector.sql.pool.commons.CommonsConnectionPool;
+import org.eclipse.dataspaceconnector.sql.pool.commons.CommonsConnectionPoolConfig;
+import org.eclipse.dataspaceconnector.transaction.local.DataSourceResource;
+import org.eclipse.dataspaceconnector.transaction.local.LocalDataSourceRegistry;
+import org.eclipse.dataspaceconnector.transaction.local.LocalTransactionContext;
+import org.h2.jdbcx.JdbcDataSource;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class SqlAssetIndexTest {
+
+    private static final String DATASOURCE_NAME = "asset";
+
+    private SqlAssetIndex sqlAssetIndex;
+    private ConnectionPool connectionPool;
+
+    @BeforeEach
+    void setUp() throws SQLException {
+        var monitor = new ConsoleMonitor();
+        var txManager = new LocalTransactionContext(monitor);
+        DataSourceRegistry dataSourceRegistry = new LocalDataSourceRegistry(txManager);
+        var transactionContext = (TransactionContext) txManager;
+        var jdbcDataSource = new JdbcDataSource();
+        jdbcDataSource.setURL("jdbc:h2:mem:");
+
+        var connection = jdbcDataSource.getConnection();
+        var dataSource = new ConnectionFactoryDataSource(() -> connection);
+        connectionPool = new CommonsConnectionPool(dataSource, CommonsConnectionPoolConfig.Builder.newInstance().build());
+        var poolDataSource = new ConnectionPoolDataSource(connectionPool);
+        dataSourceRegistry.register(DATASOURCE_NAME, poolDataSource);
+        txManager.registerResource(new DataSourceResource(poolDataSource));
+        sqlAssetIndex = new SqlAssetIndex(dataSourceRegistry, DATASOURCE_NAME, transactionContext, new ObjectMapper(), new PostgresSqlAssetQueries());
+
+        try (var inputStream = getClass().getClassLoader().getResourceAsStream("schema.sql")) {
+            var schema = new String(Objects.requireNonNull(inputStream).readAllBytes(), StandardCharsets.UTF_8);
+            transactionContext.execute(() -> SqlQueryExecutor.executeQuery(connection, schema));
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        connectionPool.close();
+    }
+
+    @Test
+    @DisplayName("Accept an asset and a data address that don't exist yet")
+    void acceptAssetAndDataAddress_doesNotExist() {
+        var assetExpected = getAsset("id1");
+        sqlAssetIndex.accept(assetExpected, getDataAddress());
+
+        var assetFound = sqlAssetIndex.findById("id1");
```

Comment:
It can be problematic to use the System-under-test to verify the System-under-test.